File: src/main/go/easy/array/flipping_an_image.go
package array

func flipAndInvertImage(image [][]int) [][]int {
    for _, v := range image {
        length := len(v) - 1
        for i := 0; i < len(v)/2; i++ {
            tmp := v[i]
            if v[length] == 0 {
                v[i] = 1
            } else {
                v[i] = 0
            }
            if tmp == 0 {
                v[length] = 1
            } else {
                v[length] = 0
            }
            length--
        }
    }
    return image
}

File: src/main/kotlin/easy/array/FlippingAnImage.kt
package easy.array

fun flipAndInvertImage(image: Array<IntArray>): Array<IntArray> {
    val arr = Array(image.size) { intArrayOf() }

    for ((j, img) in image.withIndex()) {
        val tmp = IntArray(img.size)
        var length = img.lastIndex
        for (i in 0..img.size / 2) {
            if (img[length] == 0)
                tmp[i] = 1
            if (img[i] == 0)
                tmp[length] = 1
            length--
        }
        arr[j] = tmp
    }
    return arr
}
